src/BuildScriptGenerator/DefaultDockerfileGenerator.cs

```diff
@@ -47,90 +47,104 @@ internal class DefaultDockerfileGenerator : IDockerfileGenerator
+                    }
 
-                // If the runtime platform version wasn't previously provided, see if we can detect one from the current detected platform.
-                // Note: we first need to ensure that the current detected platform is the same as the runtime platform name previously set or pvodied.
-                if (!string.IsNullOrEmpty(dockerfileRuntimeImage) && dockerfileRuntimeImage.Equals(detectedRuntimeName, StringComparison.OrdinalIgnoreCase) &&
-                     string.IsNullOrEmpty(dockerfileRuntimeImageTag))
-                {
-                    dockerfileRuntimeImageTag = this.ConvertToRuntimeVersion(dockerfileRuntimeImage, platformDetectorResult.PlatformVersion);
+                    // If the runtime image has been set manually or by the platform detection result, stop searching.
+                    if (!string.IsNullOrEmpty(dockerfileRuntimeImage))
+                    {
+                        break;
+                    }
                 }
 
-                // If the runtime image has been set manually or by the platform detection result, stop searching.
-                if (!string.IsNullOrEmpty(dockerfileRuntimeImage))
+                var formattedCreateScriptArguments = createScriptArguments.Any() ?
+                    string.Join(' ', createScriptArguments.Select(arg => $"-{arg.Key} {arg.Value}")) : string.Empty;
+
+                var properties = new DockerfileProperties()
                 {
-                    break;
-                }
+                    RuntimeImageName = dockerfileRuntimeImage,
+                    RuntimeImageTag = dockerfileRuntimeImageTag,
+                    BuildImageName = dockerfileBuildImageName,
+                    BuildImageTag = dockerfileBuildImageTag,
+                    CreateScriptArguments = formattedCreateScriptArguments,
+                };
+
+                var generatedDockerfile = TemplateHelper.Render(
+                    TemplateHelper.TemplateResource.Dockerfile,
+                    properties,
+                    this.logger);
+
+                // Remove the Container Registry Analysis snippet, if it exists in the template.
+                var pattern = "# DisableDockerDetector \".*?\"\n";
+                generatedDockerfile = Regex.Replace(generatedDockerfile, pattern, string.Empty);
+
+                var buildEventProps = new Dictionary<string, string>()
+                {
+                    { "runtimeImageName", properties.RuntimeImageName },
+                    { "runtimeImageTag", properties.RuntimeImageTag },
+                    { "buildImageName", properties.BuildImageName },
+                    { "buildImageTag", properties.BuildImageTag },
+                    { "createScriptArguments", properties.CreateScriptArguments },
+                    { "generatedDockerfile", generatedDockerfile },
```

Comment:
Let's remove the `createScriptArguments` and `generatedDockerfile` properties from this `buildEventProps` dictionary -- they won't be useful for us and will bloat the logging.

Reply:
alright.